README.md

```diff
@@ -30,27 +51,233 @@ npm install --save ipfsd-ctl
 
 IPFS daemons are already easy to start and stop, but this module is here to do it from JavaScript itself.
 
+### Local node
+
 ```js
 // Start a disposable node, and get access to the api
-// print the node id, and kill the temporary daemon
-
-// IPFS_PATH will point to /tmp/ipfs_***** and will be
-// cleaned up when the process exits.
+// print the node id, and stop the temporary daemon
 
-var ipfsd = require('ipfsd-ctl')
+const DaemonFactory = require('ipfsd-ctl')
+const df = DaemonFactory.create()
 
-ipfsd.disposableApi(function (err, ipfs) {
-  ipfs.id(function (err, id) {
+df.spawn(function (err, ipfsd) {
+  const ipfsCtl = ipfsd.ctl
+  const ipfsCtrl = ipfsd.ctrl
+  ipfsCtl.id(function (err, id) {
     console.log(id)
-    process.exit()
+    ipfsCtrl.stopDaemon()
   })
 })
 ```
 
-If you need want to use an existing ipfs installation you can set `$IPFS_EXEC=/path/to/ipfs` to ensure it uses that.
+### Remote node
+
+```js
+// Start a remote disposable node, and get access to the api
+// print the node id, and stop the temporary daemon
+
+const DaemonFactory = require('ipfsd-ctl')
+
+const port = 9999
+const df = DaemonFactory.create({remote: true, port})
+df.start(port, (err) => {
+  if (err) {
+    throw err
+  }
+  
+  df.spawn(function (err, ipfsd) {
+    const ipfsCtl = ipfsd.ctl
+    const ipfsCtrl = ipfsd.ctrl
+    ipfsCtl.id(function (err, id) {
+      console.log(id)
+      ipfsCtrl.stopDaemon()
+      server.stop()
+    })
+  })  
+})
+```
+
+It's also possible to start the server from `.aegir` `pre` and `post` hooks. 
+
+```js
+'use strict'
+
+const server = require('./src').server
+
+module.exports = {
+  karma: {
+    files: [{
+      pattern: 'test/fixtures/**/*',
+      watched: false,
+      served: true,
+      included: false
+    }],
+    singleRun: true
+  },
+  hooks: {
+    browser: {
+      pre: server.start,
+      post: server.stop
+    }
+  }
+}
+```
+
+## API
+
+### Daemon Factory
+
+#### Create a `DaemonFactory`
+
+```js
+const DemonFactory = require('ipfsd-ctl')
+const df = DemonFactory.create([opts])
+```
```

Comment:
Let's be consistent and call this `options`.

Reply:
Bikeshedding now but `new DaemonFactory(options)`? Love them constructors 😛

Reply:
@alanshaw we had this discussion with @diasdavid a few days back and settled on a static method instead of a class instance - https://github.com/ipfs/js-ipfsd-ctl/pull/176#discussion_r156731263